synapse/rest/admin/users.py

```diff
@@ -282,13 +289,17 @@ def __init__(self, hs):
 
 class DeactivateAccountRestServlet(RestServlet):
     PATTERNS = historical_admin_path_patterns("/deactivate/(?P<target_user_id>[^/]*)")
+    PERMISSION_CODE = "USER_DEACTIVATE"
 
     def __init__(self, hs):
         self._deactivate_account_handler = hs.get_deactivate_account_handler()
         self.auth = hs.get_auth()
 
     async def on_POST(self, request, target_user_id):
-        await assert_requester_is_admin(self.auth, request)
+        authorised_by_token = await self.check_authorized_admin_token_in_use(request)
+        if not authorised_by_token:
+            await assert_requester_is_admin(self.auth, request)
```

Comment:
can we just make `check_authorized_admin_token_in_use` also do the hoop-jumping to allow requests from admin users so we don't have to repeat this boilerplate every time?

Reply:
ideally, I'd like to remove matrix users entirely from being able to call things on `/admin`, since it's both inconvenient (none of this is stuff we want to expose through clients, so you have to do the login dance as an API client, and then you use that like an opaque token anyway) and also doesn't make sense when you think of administrators not being the same as the users (e.g. hosted solutions)

Reply:
well, that's a separate discussion. In the meantime, better to have it in one place than hundreds?